File: Data Structures & Algorithms/find-pivot-index/submission-0.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        int[] pr = new int[n];
        int[] sf = new int[n];
        pr[0] = nums[0];
        sf[n - 1] = nums[n - 1];
        for(int i = n - 2; i >= 0; i--) sf[i] = nums[i] + sf[i + 1];
        for(int i = 1; i < n; i++) pr[i] = nums[i] + pr[i - 1];
        for(int i = 0; i < n; i++) if(pr[i] == sf[i]) return i;
        return -1;
    }
}

File: Data Structures & Algorithms/find-all-numbers-disappeared-in-an-array/submission-0.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        unordered_map<int,bool>mp;
        for(int i : nums) mp[i] = true;
        vector<int>ans;
        for(int i = 1; i <= nums.size(); i++) if(mp.find(i) == mp.end()) ans.push_back(i);
        return ans;
    }
};